// @kodingdotninja/use-tailwind-breakpoint/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});



var _chunkQ63M5F2Pjs = require('./chunk-Q63M5F2P.js');

// src/index.ts





var _react = require('react');
function create(screens) {
  function useBreakpoint(breakpoint, defaultValue = false) {
    const [match, setMatch] = _react.useState.call(void 0, () => defaultValue);
    const matchRef = _react.useRef.call(void 0, defaultValue);
    _chunkQ63M5F2Pjs.useIsomorphicEffect.call(void 0, () => {
      if (!(_chunkQ63M5F2Pjs.isBrowser && "matchMedia" in window))
        return void 0;
      function track() {
        var _a;
        const value = (_a = screens[breakpoint]) != null ? _a : "999999px";
        const query = window.matchMedia(`(min-width: ${value})`);
        matchRef.current = query.matches;
        if (matchRef.current != match) {
          setMatch(matchRef.current);
        }
      }
      window.addEventListener("resize", track);
      return () => window.removeEventListener("resize", track);
    });
    return match;
  }
  function useBreakpointEffect(breakpoint, effect) {
    const match = useBreakpoint(breakpoint);
    _react.useEffect.call(void 0, () => effect(match));
    return null;
  }
  function useBreakpointValue(breakpoint, valid, invalid) {
    const match = useBreakpoint(breakpoint);
    const value = _react.useMemo.call(void 0, () => match ? valid : invalid, [invalid, match, valid]);
    return value;
  }
  return {
    useBreakpoint,
    useBreakpointEffect,
    useBreakpointValue
  };
}
var src_default = create;






exports.create = create; exports.default = src_default; exports.isBrowser = _chunkQ63M5F2Pjs.isBrowser; exports.isSSR = _chunkQ63M5F2Pjs.isSSR; exports.useIsomorphicEffect = _chunkQ63M5F2Pjs.useIsomorphicEffect;
